app/components/course/CourseDropDownButton.tsx

```diff
@@ -1,41 +1,82 @@
-  };
+  const updateCourseInContext = (newState: string) => {
+    console.log(coursesSelected);
+    let to_add : SingleCourse = { courseID: courseCode, title: title, status: newState };
+    const updatedCourses = coursesSelected.map(course =>
+      course.courseID === courseCode
+        ? { ...course, status: newState }
+        : course);
+    let selectedIDS = updatedCourses.map(course => course.courseID);
 
-  const handleOption1 = () => {
-    setIsOpen(false);
-    setDropDownText("Planned");
-  };
-  const handleOption2 = () => {
-    setIsOpen(false);
-    setDropDownText("In Progress");
+    if (!selectedIDS.includes(to_add.courseID)) {
+      updatedCourses.push(to_add);
+    }
+    console.log(updatedCourses);
+    setCoursesSelected(updatedCourses);
   };
-  const handleOption3 = () => {
+  
+  const handleOption = (newStatus: string) => {
+    setDropDownText(newStatus);
+    updateCourseInContext(newStatus);
     setIsOpen(false);
-    setDropDownText("Interested");
   };
+  /*
+  useEffect(() => {
+    if (dropDownText !== "No Selection") {
+      setIsOpen(true);
+    }
+  }, [dropDownText]);
+  */
+  
 
   return (
-    <div className="relative" onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
-      <div className="text-sm font-semibold text-gray-25 bg-primary-700 px-2 py-2.5 border border-solid border-gray-300 rounded-lg cursor-pointer text-center w-28">
+    <div
+      className="relative"
+      onMouseEnter={() => setIsOpen(true)}
+      onMouseLeave={() => setIsOpen(false)}
+    >
+      <div
+        id="fixed-size-div"
+        className={`text-sm font-semibold px-2 py-2.5 border border-solid border-gray-300 rounded-lg cursor-pointer text-center ${colorMap[dropDownText]}`}
+      >
         {dropDownText}
       </div>
       {isOpen && (
         <div className="absolute w-48 bg-white shadow-lg rounded-lg border border-solid border-gray-300 z-10">
           <ul>
-            <li className="py-2 px-4 hover:bg-gray-100 cursor-pointer w-48" onClick={handleOption1}>Planned</li>
-            <li className="py-2 px-4 hover:bg-gray-100 cursor-pointer w-48" onClick={handleOption2}>In Progress</li>
-            <li className="py-2 px-4 hover:bg-gray-100 cursor-pointer w-48" onClick={handleOption3}>Interested</li>
+            {dropDownText === "No Selection" && (
+              <>
+                <li className="py-2 px-4 hover:bg-gray-100 cursor-pointer" onClick={() => handleOption("Planned")}>Planned</li>
+                <li className="py-2 px-4 hover:bg-gray-100 cursor-pointer" onClick={() => handleOption("In Progress")}>In Progress</li>
+                <li className="py-2 px-4 hover:bg-gray-100 cursor-pointer" onClick={() => handleOption("Completed")}>Completed</li>
+              </>
+            )}
+            {dropDownText !== "No Selection" && (
+              <li className="py-2 px-4 hover:bg-gray-100 cursor-pointer" onClick={() => handleOption("No Selection")}>Remove</li>
+            )}
```

Comment:
Why is it change to remove only when selected?

Reply:
I'm not quite sure what you're referring to, it may have been a change made by Xinchao which I merged into my branch.